### PR TITLE
[APM] Add setting to fully disable HTTP receiver

### PR DESCRIPTION
**Pull request by ichinaski**
### What does this PR do?

This is a follow up on https://github.com/DataDog/datadog-agent/pull/27615

Add new setting to disable the HTTP receiver in the trace-agent. This setting should almost never be disabled, and it's only a convenience parameter for Opentelemetry extensions.
Disabling the receiver is semantically equivalent to setting the receiver_port to 0 and receiver_socket to "".

Note: I was doubting whether this is useful or noisy (as it’s going to be true almost always), but I think receiver_enabled:false is clearer than setting every listener config to 0 or ""

### Motivation

https://github.com/DataDog/datadog-agent/pull/27615#pullrequestreview-2180439980

### Describe how to test/QA your changes

N/A

### CODE REVIEW

File: pkg/config/config_template.yaml [hunk 1]
```diff
@@ -1042,6 +1042,15 @@ api_key:
   #
   # env: none
 
+  ## @param receiver_enabled - bool - optional - default: true
+  ## @env DD_APM_RECEIVER_ENABLED - bool - optional - default: true
+  ## Allows to fully disable HTTP receiver in trace-agent. This setting should almost never
+  ## be disabled, and it's only a convenience parameter for Opentelemetry extensions.
```
Comment (drichards-87) [inline pkg/config/config_template.yaml:1048]:
```suggestion
  ## Allows the HTTP receiver in trace-agent to be fully disabled. This setting should almost never
  ## be disabled and is only a convenience parameter for OpenTelemetry extensions.
```

Comment (drichards-87) [reply pkg/config/config_template.yaml:1048]: Could you provide some context on what we mean by "convenience parameter"?

File: releasenotes/notes/apm-receiver-enabled-setting-5a3287952ffc8858.yaml [hunk 2]
```diff
@@ -0,0 +1,15 @@
...
+# section. This may mean repeating some details, but each section
+# must be readable independently of the other.
+#
+# Each section note must be formatted as reStructuredText.
+---
+enhancements:
+  - |
+    APM: Add new setting to disable the HTTP receiver in the
+    trace-agent. This setting should almost never be disabled, and
+    it's only a convenience parameter for Opentelemetry extensions.
```
Comment (drichards-87) [inline releasenotes/notes/apm-receiver-enabled-setting-5a3287952ffc8858.yaml:13]:
```suggestion
    trace-agent. This setting should almost never be disabled and
    is only a convenience parameter for OpenTelemetry extensions.
```

**Review by drichards-87 — APPROVED** [review]
Left some suggestions and a question from Docs + approved the PR.

**Review by songy23 — APPROVED** [review]
Could you fix this as well? https://github.com/DataDog/datadog-agent/blob/5a13ffbb8ef27a1b7fe32839fff5f6623c179fc7/pkg/trace/stats/oteltest/testutil.go#L58

**Review by ajgajg1134 — APPROVED** [review]

**Review by ogaca-dd — APPROVED** [review]
LGTM for files owned by ASC

**Review by dinooliva — APPROVED** [review]

**Pull request merged**